src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -112,7 +113,7 @@ internal class GrpcWorkerChannel : IRpcWorkerChannel, IDisposable
             _processInbound = state => ProcessItem((InboundGrpcEvent)state);
             _hostingConfigOptions = hostingConfigOptions;
 
-            _workerCapabilities = new GrpcCapabilities(_workerChannelLogger);
+            _grpcCapabilitiesFactory = grpcCapabilitiesFactory;
```

Comment:
This change isn't great as some locations check capabilities before they're set, so now they're null... I'm going to think through this. The behavior of this capabilities object is pretty fragile (and I've just realized that we check/parse the values a lot... may have to refactor this first).

Reply:
I ended up simplifying this behavior by adding a virtual method I can override to intercept these capabilities in a test. It's still not beautiful, but we have plans already to overhaul a bit of this (capabilities included) and I felt like I was getting too far into the weeds.